File: server/Steel_Analysis_API/Steel_Analysis_API/Models/Analysis.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using Newtonsoft.Json;

namespace Steel_Analysis_API.Models
{
    public class Analysis
    {
        public string name, elements, steelgrade;
        public double weight, maxWeight;
        public List<AnalysisElement> elementList;
        public List<AddedAlloy> addedAlloys = new List<AddedAlloy>();

        [JsonConstructor]
        public Analysis(string name, string steelgrade, string elements, double weight, double maxWeight)
        {
            this.name = name;
            this.steelgrade = steelgrade;
            this.elements = elements;
            this.weight = weight;
            this.maxWeight = maxWeight;
        }

        public Analysis(string name, string steelgrade, double weight, double maxWeight, List<AnalysisElement> elementList)
        {
            this.name = name;
            this.steelgrade = steelgrade;
            this.weight = weight;
            this.maxWeight = maxWeight;
            this.elementList = elementList;
        }

        public double TotalPrice
        {
            get
            {
                return addedAlloys.Sum(alloy => alloy.TotalPrice);
            }
        }

        public Analysis DeepCopy()
        {
            Analysis temp = new Analysis(name, steelgrade, weight, maxWeight, elementList);
            List<AnalysisElement> tempList = new List<AnalysisElement>();

            foreach (AnalysisElement ae in this.elementList)
            {
                tempList.Add(new AnalysisElement(ae.name, ae.min, ae.aim, ae.max, ae.actual, ae.weight));
            }

            List<AddedAlloy> tempAlloys = new List<AddedAlloy>();

            foreach (AddedAlloy alloy in addedAlloys)
            {
                AddedAlloy tempAlloy = new AddedAlloy(alloy.name, alloy.price);
                tempAlloy.Weight = alloy.Weight;
                tempAlloys.Add(tempAlloy);
            }

            temp.elementList = tempList;
            temp.addedAlloys = tempAlloys;

            return temp;
        }

        public override string ToString()
        {
            string print = "";

            print += "---------------------------------------\n";
            print += $"Total price: {TotalPrice}\n";
            foreach (AnalysisElement ae in elementList)
            {
                print += ae.ToString() + "\n";
            }

            print += "---------------------------------------";
            print += "ADDED ALLOYS\n";

            foreach (AddedAlloy alloy in addedAlloys)
            {
                print += alloy.ToString() + "\n";
            }

            print += "---------------------------------------";

            return print;
        }
    }
}

File: client/src/components/Calculate/Analysis/Analysis.js
import React from 'react'
import './Analysis.css'
const regex = /\B(?=(\d{3})+(?!\d))/g

export default function Analysis (props) {
  return (
    <div id='analysis-container'>
      <h3>{props.title}</h3>
      <label>Steelgrade: {props.analysis.steelgrade}</label>
      <label>
        Weight:{' '}
        {props.analysis.weight
          .toFixed(1)
          .toString()
          .replace(regex, ' ')}
        kg
      </label>
      <label>
        Max weight: {props.analysis.maxWeight.toString().replace(regex, ' ')}kg
      </label>

      <div className='element-table'>
        <table>
          <thead>
            <tr>
              <th>Name</th>
              <th>Actual</th>
              <th>Min</th>
              <th>Aim</th>
              <th>Max</th>
            </tr>
          </thead>
          <tbody>
            {props.analysis.elementList.map(e => {
              return (
                <tr>
                  <td>{e.name}</td>
                  <td className='actual-analysis'>
                    {(e.actual * 100).toFixed(2)}%
                  </td>
                  <td>{(e.min * 100).toFixed(2)}%</td>
                  <td>{(e.aim * 100).toFixed(2)}%</td>
                  <td>{(e.max * 100).toFixed(2)}%</td>
                </tr>
              )
            })}
          </tbody>
        </table>
      </div>
    </div>
  )
}
